lib/java_buildpack/framework/opentelemetry_javaagent.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2023 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+
+module JavaBuildpack
+  module Framework
+
+    # Main class for adding the OpenTelemetry Javaagent instrumentation
+    class OpenTelemetryJavaagent < JavaBuildpack::Component::VersionedDependencyComponent
```

Comment:
Running unit tests throws errors relating to the name of the framework - I think this is because the class name casing does not match the framework/spec/config file names. There should be an underscore between changes in case, i.e. keeping the class name 'OpenTelemetryJavaagent' would require the framework file to be called 'open_telemetry_javaagent'. I tested the reverse of this - i.e. renaming the class & framework entry in config/components.yml to 'OpentelemetryJavaagent' and this works.

Reply:
Renamed all the files accordingly.